### PR TITLE
added urllib and replaced replaceSpecialCharactersInPlaylist redundat code with parser.quote_plus | github actions autobuild

### PR DESCRIPTION
**Pull request by Likosdev**
Hi There,

I remember us discussing this in Discord and propose this fix for that very long function.
I actually did not test this though :P

### CODE REVIEW

**Comment by Likosdev** [comment]
note: if you can provide me with some example files i am happy to test this further

**Comment by Likosdev** [comment]
here it is, the autobuild github action :3

**Comment by Schallbert** [comment]
urllib should do the trick. 
Shall we try to setup a CI? 
I could create some dummy playlists and write a couple of  integration tests.

**Comment by Likosdev** [comment]
CI sounds good!

File: .github/workflows/generate_exe.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    - uses: actions/checkout@v2
+    - name: PyInstaller Windows
+      uses: JackMcKew/pyinstaller-action-windows@main
+      with:
+        path: ./
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: AutoStructureFilesForDFplayer
+        path: ./dist/windows # or path/to/artifact
```
Comment (Schallbert) [inline .github/workflows/generate_exe.yml:24]: As I'd like to have the executable in one file only, please add `--onefile` to the pyinstaller command

File: AutoStructureFilesForDFplayer.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 import sys
 import os.path
 import shutil
+# urllib should do what you need in replaceSpecialCharactersInPlaylist
```
Comment (Schallbert) [inline AutoStructureFilesForDFplayer.py]: re-formulate comment, e.g. "use urllib to escape special (UTF8) characters in filenames"

File: AutoStructureFilesForDFplayer.py [hunk 3]
```diff
@@ -18,92 +20,7 @@ def playerFSnumbers(indx, maxCnt):
...
-    line_of_playlist = line_of_playlist.replace("%C3%B8", "ø")
-    line_of_playlist = line_of_playlist.replace("%C3%B9", "ù")
-    line_of_playlist = line_of_playlist.replace("%C3%BA", "ú")
-    line_of_playlist = line_of_playlist.replace("%C3%BB", "û")
-    line_of_playlist = line_of_playlist.replace("%C3%BC", "ü")
-    line_of_playlist = line_of_playlist.replace("%C3%BD", "ý")
-    line_of_playlist = line_of_playlist.replace("%C3%BE", "þ")
-    line_of_playlist = line_of_playlist.replace("%C3%BF", "ÿ")
-    return line_of_playlist
+    return quote_plus(line_of_playlist)
```
Comment (Schallbert) [inline AutoStructureFilesForDFplayer.py:23]: Yeah, this is much more short and concise :)

File: AutoStructureFilesForDFplayer.spec [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# -*- mode: python ; coding: utf-8 -*-
+
+block_cipher = None
+
+
+a = Analysis(['AutoStructureFilesForDFplayer.py'],
```
Comment (Schallbert) [inline AutoStructureFilesForDFplayer.spec:6]: My first project in Python. Is it common/usual to have the spec attached to the source code?

**Review by Schallbert — CHANGES_REQUESTED** [review]
Looks much more clean now. Would need some help with the CI though.

File: .github/workflows/generate_exe.yml (hunk 1 above)
Comment (Likosdev) [reply .github/workflows/generate_exe.yml:24]: this spec file is generated with the `--one-file` option.  

https://pyinstaller.readthedocs.io/en/stable/spec-files.html for reference.

File: AutoStructureFilesForDFplayer.spec (hunk 4 above)
Comment (Likosdev) [reply AutoStructureFilesForDFplayer.spec:6]: its like a make file basically:
https://pyinstaller.readthedocs.io/en/stable/spec-files.html

**Comment by Likosdev** [comment]
I just checked, the provided spec-file and the build artifacts, its built as one exe file
comment changed according to your suggestion

**Review by Schallbert — APPROVED** [review]
LGTM

**Pull request merged**